cmd/kube-proxy/app/server.go

```diff
@@ -914,30 +915,75 @@ func (s *ProxyServer) birthCry() {
 //
 // The order of precedence is:
 //  1. if bindAddress is not 0.0.0.0 or ::, then it is used as the primary IP.
-//  2. if the Node object can be fetched, then its primary IP is used as the primary IP
-//     (and its secondary IP, if any, is just ignored).
-//  3. otherwise the primary node IP is 127.0.0.1.
+//  2. if the Node object can be fetched, then its address(es) is/are used
+//  3. otherwise the node IPs are 127.0.0.1 and ::1
 //
 // In all cases, the secondary IP is the zero IP of the other IP family.
 func detectNodeIPs(client clientset.Interface, hostname, bindAddress string) (v1.IPFamily, map[v1.IPFamily]net.IP) {
-	nodeIP := netutils.ParseIPSloppy(bindAddress)
-	if nodeIP.IsUnspecified() {
-		nodeIP = utilnode.GetNodeIP(client, hostname)
+	primaryFamily := v1.IPv4Protocol
+	nodeIPs := map[v1.IPFamily]net.IP{
+		v1.IPv4Protocol: net.IPv4(127, 0, 0, 1),
+		v1.IPv6Protocol: net.IPv6loopback,
 	}
-	if nodeIP == nil {
-		klog.InfoS("Can't determine this node's IP, assuming 127.0.0.1; if this is incorrect, please set the --bind-address flag")
-		nodeIP = netutils.ParseIPSloppy("127.0.0.1")
+
+	if ips := GetNodeIPs(client, hostname); len(ips) > 0 {
```

Comment:
before we only did a call to the apiserver if the bindaddress as unspecified, we are doing this call always now and this will add some delay on the bootrap (unless you use LocalModeNodeCIDR that you have to wait for the node nevertheless)

Should we worry about this?

Reply:
Yeah, I noticed this when I reviewed it before. But kube-proxy doesn't actually do anything useful until it's able to talk to the apiserver anyway.

One possible catch: the healthz server doesn't start until after this point, so this will delay kube-proxy readiness if you have a readinessProbe on it...